theano/tensor/nlinalg.py

```diff
@@ -144,6 +144,12 @@ class AllocDiag(Op):
 
     __props__ = ()
 
+    def __init__(self):
+        warnings.warn("DeprecationWarning: theano.tensor.nlinalg.AllocDiag"
+                      "is deprecated, please use theano.tensor.Diag"
```

Comment:
I would put the warning inside the make_node, as otherwise, this will always warn, as this file instantiate one such object. Idem for the other warning.